app/controllers/category_controller.rb

```diff
@@ -36,8 +39,22 @@ def create
   def delete
     anchor = @category.parent_id.present? ? "#category-#{@category.parent_id}" : nil
     @category.destroy #_but_attach_children_to_parent
+    
     flash[:notice] = "Category has been deleted"
     redirect_to "#{request.env['HTTP_REFERER']}#{anchor}"
   end
-
+  def enable_gis
+    @category.update_attribute(:gis_enabled, true)
+    @category.descendants.each {|d| d.update_attribute(:gis_enabled, true)}
+    
+    flash[:notice] = "GIS Enabled for #{@category.title}"
```

Comment:
Can we have the success notice mention the children, to make this explicit?  Something like 
`GIS Enabled for #{@category.title} and #{@category.descendants.count} child categories` or `GIS Enabled for #{@category.title} and all children` would make the cascading behavior clearer to users.